nifi-registry-core/nifi-registry-web-ui/src/main/webapp/services/nf-registry.api.js

```diff
@@ -75,6 +75,137 @@ NfRegistryApi.prototype = {
         );
     },
 
+    /**
+     * Retrieves the specified versioned flow snapshot for an existing droplet the registry has stored.
+     *
+     * @param {string}  dropletUri      The uri of the droplet to request.
+     * @param {number}  versionNumber   The version of the flow to request.
+     * @returns {*}
+     */
+    exportDropletVersionedSnapshot: function (dropletUri, versionNumber) {
+        var self = this;
+        var url = '../nifi-registry-api/' + dropletUri + '/versions/' + versionNumber + '/export';
+        var options = {
+            headers: headers,
+            observe: 'response'
+        };
+
+        return this.http.get(url, options).pipe(
+            map(function (response) {
+                // export the VersionedFlowSnapshot by creating a hidden anchor element
+                var stringSnapshot = encodeURIComponent(JSON.stringify(response.body, null, 2));
```

Comment:
Is it necessary to call `JSON.stringify()` or is it possible to just get the text content of response.body?  If `response.body` is already JSON, that means it went through an unnecessary serialization step, which would be great to avoid if possible.

Reply:
@exceptionfactory Good catch! I can return a `responseType: 'text'` and omit calling `JSON.stringify()`. The file content will not be in pretty print format but that seems ok to me.